### PR TITLE
Allow Custom Visual Props for Custom Chart

### PR DESCRIPTION
**Pull request by divyamlamiyan**
This commit 3fb281f64e8bc7beb403d171aa19bb78dbf7e8a1.

### CODE REVIEW

File: example/custom-bar-chart/custom-chart.ts [hunk 1]
```diff
@@ -21,10 +21,21 @@ import {
...
 import _ from 'lodash';
 
+Chart.register(ChartDataLabels);
+
 let globalChartReference: Chart;
 
+const color = ['red', 'green', 'blue'];
+
+const indexCol = {
+    0: 'color',
```
Comment (chetan1507) [inline example/custom-bar-chart/custom-chart.ts:34]: can we add more meaning ful names here?

File: example/custom-bar-chart/custom-chart.ts [hunk 2]
```diff
@@ -21,10 +21,21 @@ import {
...
+import ChartDataLabels from 'chartjs-plugin-datalabels';
 import _ from 'lodash';
 
+Chart.register(ChartDataLabels);
+
 let globalChartReference: Chart;
 
+const color = ['red', 'green', 'blue'];
+
+const indexCol = {
```
Comment (chetan1507) [inline example/custom-bar-chart/custom-chart.ts:33]: visualPropKeyMap

File: example/custom-bar-chart/custom-chart.ts [hunk 3]
```diff
@@ -242,6 +280,37 @@ const renderChart = async (ctx: CustomChartContext): Promise<void> => {
...
+                    values: ['red', 'green', 'yellow'],
+                    label: 'Colors',
+                },
+                {
+                    type: 'section',
+                    key: 'accordion',
+                    label: 'Accordion',
+                    children: [
+                        {
+                            key: 'Color2',
```
Comment (chetan1507) [inline example/custom-bar-chart/custom-chart.ts:298]: are we using this? if not lets remove

File: example/custom-bar-chart/custom-chart.ts [hunk 4]
```diff
@@ -116,6 +137,23 @@ function render(ctx: CustomChartContext) {
             },
             options: {
                 scales: dataModel.getScales(),
+                plugins: {
+                    // Change options for ALL labels of THIS CHART
+                    datalabels: {
+                        display: allowLabels,
+                        color: 'blue',
```
Comment (chetan1507) [inline example/custom-bar-chart/custom-chart.ts:144]: would be good to add configurations for these too

**Review by chetan1507 — CHANGES_REQUESTED** [review]
added comments. please update and send again.

File: src/types/chart-to-ts-event.types.ts [hunk 5]
```diff
@@ -80,7 +80,9 @@ export interface ChartToTSEventsPayloadMap {
  *
  * @group Chart to ThoughtSpot Events
  */
-type UpdateVisualPropsEventPayload = VisualProps;
+interface UpdateVisualPropsEventPayload {
+    visualProps: VisualProps;
```
Comment (chetan1507) [inline src/types/chart-to-ts-event.types.ts:84]: this should have a corresponding contract change as well.

File: example/custom-bar-chart/custom-chart.ts (hunk 3 above)
Comment (divyamlamiyan) [reply example/custom-bar-chart/custom-chart.ts:298]: Yes we are using it.  It is used to show form builder currently.

**Pull request merged**